package.json

```diff
@@ -106,10 +106,10 @@
                     "default": "",
                     "markdownDescription": "%AWS.configuration.description.samcli.location%"
                 },
-                "aws.samcli.debug.attach.timeout.millis": {
+                "aws.samcli.lambda.timeout": {
```

Comment:
+1 for changing this, but we need to recognize that we are actually "removing" the old setting and adding a new one. Besides the improved name, this is also useful because the new default is probably better for most users: anyone that adjusted the old setting might have set it too low.

Reply:
Ah very true. I'll add some code to read the old setting to migrate users

Reply:
Sure. Though it's not a blocker imo

Reply:
Agreed. Now that I think about it though, a generic migration method on `SettingsConfiguration` would be very useful and reduce the friction of changing setting interfaces between toolkit versions. It would cover all the different ways a user might have their settings configured (i.e. Workspace/User/Global) and perform transformations to the new setting if necessary. I'll work on this as a little side project.